client/asset/dcr/spv_mixing.go

```diff
@@ -13,124 +10,29 @@ const (
 	smalletCSPPSplitPoint = 1 << 18 // 262144
 	mixedAccountName      = "mixed"
 	mixedAccountBranch    = udb.InternalBranch
-	tradingAccount        = "dextrading"
+	tradingAccountName    = "dextrading"
 )
 
-func (w *spvWallet) updateCSPPServerConfig(serverAddress string, tlsConfig *tls.Config) {
-	if serverAddress == w.csppServer && tlsConfig.ServerName == w.csppTLSConfig.ServerName &&
-		tlsConfig.RootCAs.Equal(w.csppTLSConfig.RootCAs) {
-		return // nothing to update
-	}
-
-	w.csppMtx.Lock()
-	defer w.csppMtx.Unlock()
-
-	// Stop the mixer if it is currently running. The next run will use the
-	// updated server address and tlsConfig.
-	if w.cancelCSPPMixer != nil {
-		w.cancelCSPPMixer()
-		w.cancelCSPPMixer = nil
-	}
-
-	w.csppServer = serverAddress
-	w.csppTLSConfig = tlsConfig
-}
-
-func (w *spvWallet) isMixerEnabled() bool {
-	w.csppMtx.Lock()
-	defer w.csppMtx.Unlock()
-	return w.csppServer != ""
-}
-
-// isMixing is true if the wallet is currently mixing funds.
-func (w *spvWallet) isMixing() bool {
-	w.csppMtx.Lock()
-	defer w.csppMtx.Unlock()
-	return w.cancelCSPPMixer != nil
-}
-
-// startFundsMixer starts mixing funds. Requires the wallet to be unlocked.
-func (w *spvWallet) startFundsMixer(ctx context.Context) error {
-	w.csppMtx.Lock()
-	defer w.csppMtx.Unlock()
-
-	if w.cancelCSPPMixer != nil {
-		return fmt.Errorf("funds mixer is already running")
-	}
-
-	if w.csppServer == "" {
-		return fmt.Errorf("funds mixer is disabled, configure first")
-	}
-
-	mixedAccount, err := w.dcrWallet.AccountNumber(ctx, mixedAccountName)
+func (w *spvWallet) mix(ctx context.Context, cfg *mixingConfig) {
+	mixedAccount, err := w.AccountNumber(ctx, mixedAccountName)
 	if err != nil {
-		return fmt.Errorf("unable to look up mixed account: %v", err)
+		w.log.Errorf("unable to look up mixed account: %v", err)
+		return
 	}
 
 	// unmixed account is the default account
 	unmixedAccount := uint32(defaultAcct)
 
-	ctx, cancel := context.WithCancel(ctx)
-	w.cancelCSPPMixer = cancel
+	w.log.Debugf("Starting cspp funds mixer with %s", cfg.server)
 
-	csppServer, csppTLSConfig := w.csppServer, w.csppTLSConfig
-	dialCSPPServer := func(ctx context.Context, network, addr string) (net.Conn, error) {
-		conn, err := new(net.Dialer).DialContext(ctx, network, addr)
-		if err != nil {
-			return nil, err
-		}
-		conn = tls.Client(conn, csppTLSConfig)
-		return conn, nil
+	// Don't perform any actions while transactions are not synced
+	// through the tip block.
+	if !w.spv.Synced() {
+		w.log.Debugf("Skipping autobuyer actions: transactions are not synced")
```

Comment:
Is `autobuyer` correct? Should say mixer? the autobuyer is for automatic ticket purchases correct?